src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIClient.java

```diff
@@ -33,32 +36,39 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 /**
  * An abstract class to call Rest API.
  */
 public abstract class RestAPIClient {
   private static final Logger LOG = LoggerFactory.getLogger(RestAPIClient.class);
+
   /**
    * Executes the Rest API request and returns the response.
    *
    * @param request the Rest API request
    * @return an instance of RestAPIResponse object.
    */
-  public RestAPIResponse executeGet(RestAPIRequest request) {
+  public RestAPIResponse executeGet(RestAPIRequest request) throws IOException {
     HttpGet httpGet = new HttpGet(request.getUrl());
     request.getHeaders().entrySet().forEach(e -> httpGet.addHeader(e.getKey(), e.getValue()));
-    RestAPIResponse apiResponse = null;
 
     try (CloseableHttpClient httpClient = HttpClientBuilder.create().build()) {
       try (CloseableHttpResponse httpResponse = httpClient.execute(httpGet)) {
-        apiResponse = RestAPIResponse.parse(httpResponse, request.getResponseHeaders());
+        try {
+          return RestAPIResponse.parse(httpResponse, request.getResponseHeaders());
+        } catch (RetryableException e) {
```

Comment:
This catch and rethrow is unnecessary.

Reply:
Now I see the logic better... why you want to wrap all exceptions into IOException besides the retryable one? What exception does the `.parse` method may throw?